app/assets/javascripts/views/courses.js

```diff
@@ -101,5 +111,27 @@ Yacs.views.courses = function (data) {
       c.classList[isSelected ? 'remove' : 'add']('selected');
     });
     if (isCourseSelected(c)) c.classList.add('selected');
+
+    var desc = c.querySelector('course-description');
+    var showButton = c.querySelector('show-hide-button');
+    Yacs.on('click', showButton, function (showButton, event) {
+      var descTruncated = desc.classList.contains("truncated");
+      desc.classList[descTruncated ? 'remove' : 'add']("truncated");
+      showButton.classList[descTruncated ? 'add' : 'remove']("open");
+      event.stopPropagation();
+    });
+    requireTruncation(desc, showButton);
+  });
+
+  var resizingFxn;
+  window.addEventListener("resize", function () {
+    clearTimeout(resizingFxn);
+    resizingFxn = setTimeout(function () {
```

Comment:
To prevent the function from being called too many times if the user is say resizing the window by dragging. Without the timeout, the forEach loop will be called dozens of times in a second.